xrpl/utils/txn_parser/utils/order_book_parser.py

```diff
@@ -0,0 +1,217 @@
+            # it easier to return the changed amount.
+            final_amount["value"] = value
+            # From now on you could consider final_amount as changed_amount.
+            return final_amount
+    return None
+
+
+def _get_quality(
+    taker_gets: CurrencyAmount,
+    taker_pays: CurrencyAmount,
+    book_directory: str,
+) -> str:
+    quality_hex = book_directory[-16:]  # last 16 characters of the BookDirectory
+    taker_gets_currency = taker_gets["currency"]
+    taker_pays_currency = taker_pays["currency"]
+    mantissa = float.fromhex(quality_hex[2:])
+    offset = int(quality_hex[:2], 16) - 100
+    scientific_quality = "{:.16f}e{}".format(mantissa, offset)
+    quality = Decimal(scientific_quality)
+    if taker_gets_currency == "XRP" or taker_pays_currency == "XRP":
+        quality = quality * 1000000
+    return str(quality.normalize())
+
+
+def _get_optional_fields(
+    node: NormalizedNode,
+    field_name: str,
+) -> Optional[Any]:
+    new_fields = node.get("NewFields")
+    final_fields = node.get("FinalFields")
+    if new_fields is not None:
+        return new_fields.get(field_name)
+    if final_fields is not None:
+        return final_fields.get(field_name)
+    return None
+
+
+def _get_fields(
+    node: NormalizedNode,
+    field_name: str,
+) -> Optional[Any]:
+    new_fields = node.get("NewFields")
+    final_fields = node.get("FinalFields")
+    if new_fields is not None:
+        return new_fields.get(field_name)
+    if final_fields is not None:
+        return final_fields.get(field_name)
+    return None
+
+
+def _get_offer_change(node: NormalizedNode) -> Optional[AccountOfferChange]:
+    status = _get_offer_status(node)
+    taker_gets = _get_change_amount(node, status, "TakerGets")
+    taker_pays = _get_change_amount(node, status, "TakerPays")
+    account = _get_fields(node, "Account")
+    sequence = _get_fields(node, "Sequence")
+    book_directory = _get_fields(node, "BookDirectory")
+    # if required fields are None return None
+    if (
+        taker_gets is None
+        or taker_pays is None
+        or account is None
+        or sequence is None
+        or book_directory is None
+    ):
+        return None
+    expiration_time = _get_fields(node, "Expiration")
+    flags = _get_fields(node, "Flags")
+    direction: Literal["buy", "sell"] = (
+        "sell" if (flags is not None and flags & LSF_SELL != 0) else "buy"
+    )
+    quality = _get_quality(taker_gets, taker_pays, book_directory)
+    quantity = taker_pays if direction == "buy" else taker_gets
+    total_price = taker_gets if direction == "buy" else taker_pays
+    offer_change = OfferChange(
+        direction=direction,
+        quantity=quantity,
+        total_price=total_price,
+        sequence=sequence,
+        status=status,
+        maker_exchange_rate=quality,
+    )
+    if expiration_time is not None:
+        offer_change["expiration_time"] = ripple_time_to_posix(expiration_time)
```

Comment:
I would avoid making this conversion from ripple time to posix time since it's not obvious that it's happening behind the scenes. (I expected it to just be a copy of the expiration time from the ripple data when reading the test results)